src/utils.js

```diff
@@ -95,117 +108,127 @@ function getOutfilePath(outDir, outFileName) {
+  const exists = await isFile(filePath);
 
   if (!exists) {
-    throw new Error(`Invalid params: '${filePath}' is not a file or does not exist.`)
+    throw new Error(`Invalid params: '${filePath}' is not a file or does not exist.`);
   }
 
-  return true
+  return true;
 }
 
 /**
  * Validates a directory path.
  * Throws on validation failure
- * 
+ *
  * @param {string} dirPath - The directory path to validate
  * @returns {boolean} - True if validation succeeded
  */
 async function validateDirPath(dirName, createDir) {
 
-  const exists = await isDirectory(dirName, createDir)
+  const exists = await isDirectory(dirName, createDir);
 
   if (!exists) {
-    throw new Error(`Invalid params: '${dirName}' is not a directory or could not be created.`)
+    throw new Error(`Invalid params: '${dirName}' is not a directory or could not be created.`);
   }
 
-  return true
+  return true;
 }
 
 /**
- * Checks whether the given path string resolves to an existing directory, or 
- * if a directory was created.
- * @param {string} p - The path string to check
+ * Checks whether the given path string resolves to an existing directory, and
+ * optionally creates the directory if it doesn't exist.
+ *
+ * @param {string} pathString - The path string to check
  * @param {boolean} createDir - Whether to create the directory if it doesn't exist
  * @returns {boolean} - Whether the given path is an existing directory
  */
-function isDirectory(p, createDir) {
-  return new Promise((resolve, _) => {
-    fs.stat(p, (err, stats) => {
-      if (err || !stats) {
-        if (err.code === 'ENOENT') {
-          if (!createDir) return resolve(false)
-          try {
-            fs.mkdirSync(p)
-            return resolve(true)
-          } catch (err) {
-            logError(`Directory '${p}' could not be created.`, err)
-            process.exit(1)
-          }
-        }
-        return resolve(false)
+async function isDirectory(pathString, createDir) {
+  try {
+    const stats = await fs.stat(pathString);
+    return stats.isDirectory();
+  } catch (error) {
+    if (error.code === 'ENOENT') {
+      if (!createDir) {
+        return false;
+      }
+      try {
+        await fs.mkdir(pathString);
+        return true;
+      } catch (mkdirError) {
+        logError(`Directory '${pathString}' could not be created.`, mkdirError);
+        process.exit(1);
       }
-      resolve(stats.isDirectory())
-    })
-  })
+    }
+    return false;
+  }
 }
 
 /**
  * Checks whether the given path string resolves to an existing file.
- * @param {string} p - The path string to check
+ *
+ * @param {string} pathString - The path string to check
  * @returns {boolean} - Whether the given path is an existing file
  */
-function isFile(p) {
-  return new Promise((resolve, _) => {
-    fs.stat(p, (err, stats) => {
-      if (err || !stats) {
-        return resolve(false)
-      }
-      resolve(stats.isFile())
-    })
-  })
+function isFile(pathString) {
```

Comment:
Oh, this has to be an `async` function now. Is linting otherwise passing locally?